The job **`x86_64-gnu-llvm-16`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/8060016870/job/22015377030) [(plain)](https://github.com/rust-lang/rust/commit/4dabbcb23b0dc605cdb4aa6f3499d0f053f7b600/checks/22015377030/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
GITHUB_ACTION=__run_7
GITHUB_ACTIONS=true
GITHUB_ACTION_REF=
GITHUB_ACTION_REPOSITORY=
GITHUB_ACTOR=erikdesjardins
GITHUB_API_URL=https://api.github.com
GITHUB_BASE_REF=master
GITHUB_ENV=/home/runner/work/_temp/_runner_file_commands/set_env_f280974d-38a9-4d4d-a47d-1f068feb06e9
GITHUB_EVENT_NAME=pull_request
---
GITHUB_SERVER_URL=https://github.com
GITHUB_SHA=d3bd4570902aadc8a626a43c590dd0366bea819e
GITHUB_STATE=/home/runner/work/_temp/_runner_file_commands/save_state_f280974d-38a9-4d4d-a47d-1f068feb06e9
GITHUB_STEP_SUMMARY=/home/runner/work/_temp/_runner_file_commands/step_summary_f280974d-38a9-4d4d-a47d-1f068feb06e9
GITHUB_TRIGGERING_ACTOR=erikdesjardins
GITHUB_WORKFLOW_REF=rust-lang/rust/.github/workflows/ci.yml@refs/pull/121668/merge
GITHUB_WORKFLOW_SHA=d3bd4570902aadc8a626a43c590dd0366bea819e
GITHUB_WORKSPACE=/home/runner/work/rust/rust
GOROOT_1_19_X64=/opt/hostedtoolcache/go/1.19.13/x64
---
#12 writing image sha256:a55834afc46e2738063a4a4ea9cba92d568114cc2f665ed5dc94da1d5c167128 done
#12 naming to docker.io/library/rust-ci done
#12 DONE 10.1s
##[endgroup]
Setting extra environment values for docker:  --env ENABLE_GCC_CODEGEN=1 --env GCC_EXEC_PREFIX=/usr/lib/gcc/
[CI_JOB_NAME=x86_64-gnu-llvm-16]
##[group]Clock drift check
  local time: Tue Feb 27 05:21:12 UTC 2024
  network time: Tue, 27 Feb 2024 05:21:12 GMT
  network time: Tue, 27 Feb 2024 05:21:12 GMT
##[endgroup]
sccache: Starting the server...
##[group]Configure the build
configure: processing command line
configure: 
configure: build.configure-args := ['--build=x86_64-unknown-linux-gnu', '--llvm-root=/usr/lib/llvm-16', '--enable-llvm-link-shared', '--set', 'rust.thin-lto-import-instr-limit=10', '--set', 'change-id=99999999', '--enable-verbose-configure', '--enable-sccache', '--disable-manage-submodules', '--enable-locked-deps', '--enable-cargo-native-static', '--set', 'rust.codegen-units-std=1', '--set', 'dist.compression-profile=balanced', '--dist-compression-formats=xz', '--set', 'build.optimized-compiler-builtins', '--disable-dist-src', '--release-channel=nightly', '--enable-debug-assertions', '--enable-overflow-checks', '--enable-llvm-assertions', '--set', 'rust.verify-llvm-ir', '--set', 'rust.codegen-backends=llvm,cranelift,gcc', '--set', 'llvm.static-libstdcpp', '--enable-new-symbol-mangling']
configure: target.x86_64-unknown-linux-gnu.llvm-config := /usr/lib/llvm-16/bin/llvm-config
configure: llvm.link-shared     := True
configure: rust.thin-lto-import-instr-limit := 10
configure: change-id            := 99999999
---
##[endgroup]
Testing GCC stage1 (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
   Compiling y v0.1.0 (/checkout/compiler/rustc_codegen_gcc/build_system)
    Finished release [optimized] target(s) in 1.24s
     Running `/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-codegen/x86_64-unknown-linux-gnu/release/y test --use-system-gcc --use-backend gcc --out-dir /checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/cg_gcc --release --no-default-features --mini-tests --std-tests`
Using system GCC
Using system GCC
[BUILD] example
[AOT] mini_core_hello_world
/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/cg_gcc/mini_core_hello_world
abc
---

10 LL | enum UninhabitedVariantSpace {
11    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
12 
- error: abi: ScalarPair(Initialized { value: Int(I64, false), valid_range: 0..=1 }, Initialized { value: Int(I64, false), valid_range: 0..=18446744073709551615 })
+ error: abi: ScalarPair(Initialized { value: Int(I32, false), valid_range: 0..=1 }, Initialized { value: Int(I64, false), valid_range: 0..=18446744073709551615 })
15    |
15    |
16 LL | enum ScalarPairDifferingSign {

The actual stderr differed from the expected stderr.
Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/layout/enum/enum.stderr
To update references, rerun the tests and pass the `--bless` flag
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args layout/enum.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: RUSTC_ICE="0" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/layout/enum.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=i686-unknown-linux-gnu" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/layout/enum" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/i686-unknown-linux-gnu/native/rust-test-helpers" "-Clinker=x86_64-linux-gnu-gcc" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/layout/enum/auxiliary"
stdout: none
--- stderr -------------------------------
error: align: AbiAndPrefAlign { abi: Align(2 bytes), pref: Align(8 bytes) }
   |
   |
LL | enum UninhabitedVariantAlign { //~ERROR: abi: Align(2 bytes)

error: size: Size(16 bytes)
##[error]  --> /checkout/tests/ui/layout/enum.rs:15:1
   |
   |
LL | enum UninhabitedVariantSpace { //~ERROR: size: Size(16 bytes)


error: abi: ScalarPair(Initialized { value: Int(I32, false), valid_range: 0..=1 }, Initialized { value: Int(I64, false), valid_range: 0..=18446744073709551615 })
   |
   |
LL | enum ScalarPairDifferingSign { //~ERROR: abi: ScalarPair

error: aborting due to 3 previous errors
------------------------------------------

```

</details>

--- a/compiler/rustc_abi/src/layout.rs
+++ b/compiler/rustc_abi/src/layout.rs
@@ -813,7 +813,7 @@ where
                     break;
                 }
             };
-            if let Some((old_common_prim, common_offset)) = common_prim {
+            if let Some((old_prim, common_offset)) = common_prim {
                 // All variants must be at the same offset
                 if offset != common_offset {
                     common_prim = None;
@@ -822,12 +822,12 @@ where
                 // This is pretty conservative. We could go fancier
                 // by realising that (u8, u8) could just cohabit with
                 // u16 or even u32.
-                let new_common_prim = match (old_common_prim, prim) {
+                let new_prim = match (old_prim, prim) {
                     // Allow all identical primitives.
-                    (x, y) if x == y => Some(x),
+                    (x, y) if x == y => x,
                     // Allow integers of the same size with differing signedness.
                     // We arbitrarily choose the signedness of the first variant.
-                    (p @ Primitive::Int(x, _), Primitive::Int(y, _)) if x == y => Some(p),
+                    (p @ Primitive::Int(x, _), Primitive::Int(y, _)) if x == y => p,
                     // Allow integers mixed with pointers of the same layout.
                     // We must represent this using a pointer, to avoid
                     // roundtripping pointers through ptrtoint/inttoptr.
@@ -835,22 +835,15 @@ where
                     | (i @ Primitive::Int(..), p @ Primitive::Pointer(_))
                         if p.size(dl) == i.size(dl) && p.align(dl) == i.align(dl) =>
                     {
-                        Some(p)
+                        p
                     }
-                    _ => None,
-                };
-                match new_common_prim {
-                    Some(new_prim) => {
-                        // Primitives are compatible.
-                        // We may be updating the primitive here, for example from int->ptr.
-                        common_prim = Some((new_prim, common_offset));
-                    }
-                    None => {
-                        // Primitives are incompatible.
+                    _ => {
                         common_prim = None;
                         break;
                     }
-                }
+                };
+                // We may be updating the primitive here, for example from int->ptr.
+                common_prim = Some((new_prim, common_offset));
             } else {
                 common_prim = Some((prim, offset));
             }

--- a/tests/ui/layout/enum.rs
+++ b/tests/ui/layout/enum.rs
@@ -19,6 +19,6 @@ enum UninhabitedVariantSpace { //~ERROR: size: Size(16 bytes)
 
 #[rustc_layout(abi)]
 enum ScalarPairDifferingSign { //~ERROR: abi: ScalarPair
-    A(u64),
-    B(i64),
+    A(u8),
+    B(i8),
 }

--- a/tests/ui/layout/enum.stderr
+++ b/tests/ui/layout/enum.stderr
@@ -10,7 +10,7 @@ error: size: Size(16 bytes)
 LL | enum UninhabitedVariantSpace {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-error: abi: ScalarPair(Initialized { value: Int(I64, false), valid_range: 0..=1 }, Initialized { value: Int(I64, false), valid_range: 0..=18446744073709551615 })
+error: abi: ScalarPair(Initialized { value: Int(I8, false), valid_range: 0..=1 }, Initialized { value: Int(I8, false), valid_range: 0..=255 })
   --> $DIR/enum.rs:21:1
    |
 LL | enum ScalarPairDifferingSign {

